packages/synapse-react-client/src/utils/functions/QueryFilterUtils.ts

```diff
@@ -1,24 +1,24 @@
 import {
-  ColumnModel,
   ColumnSingleValueFilterOperator,
   ColumnSingleValueQueryFilter,
   QueryFilter,
   Row,
+  SelectColumn,
 } from '@sage-bionetworks/synapse-types'
 
 export function getPrimaryKeyINFilter(
   primaryKeyColumnNames: string[],
   selectedRows: Row[],
-  columnModels: ColumnModel[],
+  selectColumns: SelectColumn[],
 ): QueryFilter {
   if (!primaryKeyColumnNames || primaryKeyColumnNames.length !== 1) {
     // If the key is undefined, then the user should have never been able to apply a filter
     // TODO: Handling a composite key would be tricky since the QueryFilter API currently only allows you to specify one column
     throw new Error('rowSelectionPrimaryKey must be defined and have length 1')
   }
   // Apply a filter that should only return the selected rows
-  const primaryKeyColumnIndex = columnModels.findIndex(
-    cm => cm.name === primaryKeyColumnNames[0],
+  const primaryKeyColumnIndex = selectColumns.findIndex(
+    sc => sc.name === primaryKeyColumnNames[0],
   )
   const filter: ColumnSingleValueQueryFilter = {
     concreteType:
```

Comment:
If columns were selected in a different order than the ColumnModels order in the Table schema, then row.values[primaryKeyColumnIndex] does not return the expected result!  Use the SelectColumns instead (since the row data is synced with the SelectColumns)